src/main/java/com/github/manolo8/darkbot/core/manager/HeroManager.java

```diff
@@ -195,6 +196,9 @@ public boolean attackMode() {
     public boolean attackMode(Npc target) {
         if (target == null) return attackMode();
         Config.ShipConfig config = this.main.config.GENERAL.OFFENSIVE;
+        if (target.npcInfo.extra.has(NpcExtra.OPPOSITE_CONFIG)) {
+            config.CONFIG = config.getConfiguration() == Configuration.FIRST ? Configuration.SECOND.ordinal() : Configuration.FIRST.ordinal();
```

Comment:
assignment is not an option, otherwise this will be saved to the config

Reply:
2 options, preference?

1. initialize new Config variable 
```
Config.ShipConfig config = new Config.ShipConfig(this.main.config.GENERAL.OFFENSIVE.CONFIG, this.main.config.GENERAL.OFFENSIVE.FORMATION);
        if (target.npcInfo.extra.has(NpcExtra.OPPOSITE_CONFIG)) {
            config.CONFIG = config.getConfiguration() == Configuration.FIRST ? Configuration.SECOND.ordinal() : Configuration.FIRST.ordinal();
        }
```

2. ternary on the config
```
return setMode(target.npcInfo.extra.has(NpcExtra.OPPOSITE_CONFIG) 
                ? config.getConfiguration() == Configuration.FIRST ? Configuration.SECOND.ordinal() : Configuration.FIRST.ordinal() 
                : config.CONFIG, target.npcInfo.attackFormation != null ?
                target.npcInfo.attackFormation : config.FORMATION);
```

Reply:
I'd just:
```
        boolean otherConfig = target.npcInfo.extra.has(NpcExtra.OPPOSITE_CONFIG);
        return setMode(
                otherConfig ? ((config.CONFIG % 2) + 1) : config.CONFIG,
                target.npcInfo.attackFormation != null ?
                    target.npcInfo.attackFormation : config.FORMATION);
```